src/igvfd/types/file_set.py

```diff
@@ -122,20 +122,30 @@ def control_for(self, request, control_for):
 
     @calculated_property(schema={
         'title': 'Submitted Files Timestamp',
-        'description': 'The timestamp the first file object in the file_set was created.',
+        'description': 'The timestamp the first file object in the file_set or associated auxiliary sets was created.',
         'comment': 'Do not submit. The timestamp is automatically calculated.',
         'type': 'string',
         'format': 'date-time',
         'notSubmittable': True
     })
-    def submitted_files_timestamp(self, request, files):
+    def submitted_files_timestamp(self, request, files, auxiliary_sets=[]):
+        timestamps = set()
         if files:
-            timestamps = set()
             for current_file_path in files:
                 file_object = request.embed(current_file_path, '@@object?skip_calculated=true')
                 timestamp = file_object.get('creation_timestamp', None)
                 if timestamp:
                     timestamps.add(timestamp)
+        if auxiliary_sets:
```

Comment:
i suggest to add files from auxiliary set and traverse the whole list of files once. 
otherwise you have code duplication lines 134-138 and 143-147

Reply:
Ok, will adjust.